### PR TITLE
Update to Wake Lock API article with latest API changes

### PR DESCRIPTION
**Pull request by tomayac**
What's changed, or what was fixed?
- Updates the article to describe the [current API](https://w3c.github.io/wake-lock/).
- New [code sample](https://wake-lock-demo.glitch.me/).

**Target Live Date:** any

- [ ] This has been reviewed and approved by (NAME)
@jpmedley PTAL (and please also review the code sample).
- [X] I have run `npm test` locally and all tests pass.
- [X] I have added the appropriate `type-something` label.
- [X] I've staged the site and manually verified that my content displays correctly.

**CC:** @petele

### CODE REVIEW

File: src/content/en/updates/2018/12/wakelock.md [hunk 1]
```diff
@@ -100,78 +95,99 @@ For example, a screen wake lock implies that the app should continue running.
...
-it’ll need to be activated first.
-
+<div class="clearfix"></div>
+Warning: We’re currently defining the permission model for `system` wake locks.
+Until this work has finished, requests for `system` wake locks will be denied immediately.
+You can track our progress in [crbug.com/985742](https://crbug.com/985742).
+
+### Get a wake lock {: #get-wake-lock }
+
+In order to request a wake lock, you need to call the `WakeLock.request()` method
```
Comment (jpmedley) [inline src/content/en/updates/2018/12/wakelock.md]: Delete 'In order'.

File: src/content/en/updates/2018/12/wakelock.md [hunk 2]
```diff
@@ -100,78 +95,99 @@ For example, a screen wake lock implies that the app should continue running.
...
+that lives on the `window` object. You pass it the desired wake lock type as the first parameter,
+which *currently* is limited to just `'screen'`. In addition to that,
```
Comment (jpmedley) [inline src/content/en/updates/2018/12/wakelock.md]: Delete 'to that'.

File: src/content/en/updates/2018/12/wakelock.md [hunk 3]
```diff
@@ -100,78 +95,99 @@ For example, a screen wake lock implies that the app should continue running.
...
+you also need a way to abort the wake lock,
+which works through the generic `AbortController` interface.
+Therefore, you first create a new
+[`AbortController`](https://developer.mozilla.org/en-US/docs/Web/API/AbortController),
+and then pass the controller’s
+[`AbortSignal`](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal)
+as the second parameter to `WakeLock.request()`.
+Two things can happen next that you need to `catch`:
+
+1. The wake lock can after a while just be regularly aborted,
```
Comment (jpmedley) [inline src/content/en/updates/2018/12/wakelock.md]: These need to be bullets. I know you counted them. But the web.dev style guide (not to mention long-standing tech writing practice) reserves numbers for procedures.

File: src/content/en/updates/2018/12/wakelock.md (hunk 1 above)
Comment (tomayac) [reply src/content/en/updates/2018/12/wakelock.md]: Done.

File: src/content/en/updates/2018/12/wakelock.md (hunk 2 above)
Comment (tomayac) [reply src/content/en/updates/2018/12/wakelock.md]: Be gone :-)

File: src/content/en/updates/2018/12/wakelock.md (hunk 3 above)
Comment (tomayac) [reply src/content/en/updates/2018/12/wakelock.md]: Makes sense.

File: src/content/en/updates/2018/12/wakelock.md [hunk 4]
```diff
@@ -100,78 +96,99 @@ For example, a screen wake lock implies that the app should continue running.
...
-### Get a wake lock object {: #get-wake-lock }
-
-In order to use the Wake Lock API, we need to create and initialize a
-`wakelock` object for the type of wake lock we want. Once created, the promise
-resolves with a `wakelock` object, but note, the wake lock isn’t active yet,
-it’ll need to be activated first.
-
+<div class="clearfix"></div>
+Warning: We’re currently defining the permission model for `system` wake locks.
+Until this work has finished, requests for `system` wake locks will be denied immediately.
```
Comment (jpmedley) [inline src/content/en/updates/2018/12/wakelock.md]: Change 'has' to 'is'.

File: src/content/en/updates/2018/12/wakelock.md [hunk 5]
```diff
@@ -100,78 +96,99 @@ For example, a screen wake lock implies that the app should continue running.
...
+you also need a way to abort the wake lock,
+which works through the generic `AbortController` interface.
+Therefore, you first create a new
+[`AbortController`](https://developer.mozilla.org/en-US/docs/Web/API/AbortController),
+and then pass the controller’s
+[`AbortSignal`](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal)
+as the second parameter to `WakeLock.request()`.
+Two things can happen next that you need to `catch`:
+
+- The wake lock can after a while just be regularly aborted,
```
Comment (jpmedley) [inline src/content/en/updates/2018/12/wakelock.md]: Comma between 'can' and 'after'.

File: src/content/en/updates/2018/12/wakelock.md [hunk 6]
```diff
@@ -100,78 +96,99 @@ For example, a screen wake lock implies that the app should continue running.
...
-
-To acquire a wake lock, we simply need to call `wakeLockObj.createRequest()`,
-which creates a new `WakeLockRequest` object. The `WakeLockRequest` object
-allows multiple components on the same page to request their own locks. The
-`WakeLock` object automatically handles the requests. The wake lock is
-released when all of the requests have been cancelled.
-
+### The wake lock lifecycle {: #wake-lock-lifecycle }
+
+When you play with the  [wake lock demo][demo], you will notice that
```
Comment (jpmedley) [inline src/content/en/updates/2018/12/wakelock.md]: Change "you will" to "you'll".

The web.dev style guide prefers contractions. Please check this throughout.

File: src/content/en/updates/2018/12/wakelock.md (hunk 4 above)
Comment (tomayac) [reply src/content/en/updates/2018/12/wakelock.md]: Done.

File: src/content/en/updates/2018/12/wakelock.md (hunk 5 above)
Comment (tomayac) [reply src/content/en/updates/2018/12/wakelock.md]: Done

File: src/content/en/updates/2018/12/wakelock.md (hunk 6 above)
Comment (tomayac) [reply src/content/en/updates/2018/12/wakelock.md]: Done

**Pull request merged**